File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function GCD($a , $b){
        if ($b == 0){
            return $a;
        }
        return $this->GCD($b , $a%$b);
    }
    function findGCD($nums) {
        return $this->GCD(max($nums) , min($nums));
    }
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def GCD(self,num1 , num2):
        if num2 ==0:
            return num1
        return self.GCD(num2,num1%num2)
    
    def findGCD(self, nums: List[int]) -> int:
        minVal = min(nums)
        maxVal = max(nums)
        return self.GCD(maxVal , minVal)